### PR TITLE
add TopNav, SignIn features

### PR DESCRIPTION
**Pull request by ilynnluoca**
@ahgood

### CODE REVIEW

File: src/redux/reducers/userSlice.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      console.log('trying async', param.inputName, param.inputPwd)
+      const response = await axios.post('https://imoogoo.com/api/v1/signin', {
+        name: param.inputName,
+        password: param.inputPwd
+      })
+      return response.data.item
+    } catch (e: any) {
+      return e.message
+    }
+  }
```
Comment (wilsonwu) [inline src/redux/reducers/userSlice.ts]: axios need to write in a js file like:
```
import axios from 'axios';

const version = '/v1';

export const postSignIn = (params) => axios.post(`${version}/signin`, params);
```

and the base url and header shoud be set in index.js like:
```
axios.defaults.baseURL = 'http://127.0.0.1:8081/api';
// axios.defaults.baseURL = 'https://imoogoo.com/api';
axios.defaults.headers.Authorization = `Bearer ${localStorage.tokenAccessToken}`;
```

File: src/redux/reducers/userSlice.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    refreshToken: null
+  }
+}
+
+export const userSignIn = createAsyncThunk(
+  'user/signIn',
+  async (param: { inputName: string | null, inputPwd: string | null }): Promise<UserState> => {
+    try {
+      param.inputName = 'admin'
+      param.inputPwd = 'aaaaaa'
```
Comment (wilsonwu) [inline src/redux/reducers/userSlice.ts]: Don't hard code the user name and password in code.

**Comment by ahgood** [comment]
It's big PR which is hard to review, If I was you, I would prefer break down this app a little bit like:
1. Create react app -> config package.json -> update basic info like readme.md -> create PR
2. Do single UI/Function -> create another PR
3. Other UI/Function changes -> create new PR

~~I downloaded this repo and installed dependencies, but `npm run start` or `npm start` didn't work for me, command line stuck at `Starting the development server...` and open `localhost:3000` just show a blank page and loading... it should be my computer issue, tested create-react-app and it also not working for me~~ it's working now, just took few mins to start, not sure why.

I also tried `npm run build` and then `cd build && npx http-server` to use a local web server to test the app, I can see most of the page, found some issues:
1. Mouse over some link like top right corner Title/logo, the pointer is not a hand, which is not a good user experience
2. Some links like project detail page, click it just show the blank page
3. Similar with above, click Project List and go to `http://localhost:3000/projects/`, the page display for 1 second and then disappear

File: src/redux/reducers/userSlice.ts (hunk 1 above)
Comment (ilynnluoca) [reply src/redux/reducers/userSlice.ts]: axios.post has already been abstracted.
axios.defaults. added

File: src/redux/reducers/userSlice.ts (hunk 2 above)
Comment (ilynnluoca) [reply src/redux/reducers/userSlice.ts]: resolve later

**Review by wilsonwu — APPROVED** [review]

**Comment by ilynnluoca** [comment]
> It's big PR which is hard to review, If I was you, I would prefer break down this app a little bit like:
> 
> 1. Create react app -> config package.json -> update basic info like readme.md -> create PR
> 2. Do single UI/Function -> create another PR
> 3. Other UI/Function changes -> create new PR
> 
> ~I downloaded this repo and installed dependencies, but `npm run start` or `npm start` didn't work for me, command line stuck at `Starting the development server...` and open `localhost:3000` just show a blank page and loading... it should be my computer issue, tested create-react-app and it also not working for me~ it's working now, just took few mins to start, not sure why.
> 
> I also tried `npm run build` and then `cd build && npx http-server` to use a local web server to test the app, I can see most of the page, found some issues:
> 
> 1. Mouse over some link like top right corner Title/logo, the pointer is not a hand, which is not a good user experience
> 2. Some links like project detail page, click it just show the blank page
> 3. Similar with above, click Project List and go to `http://localhost:3000/projects/`, the page display for 1 second and then disappear

Got it. I will break it as small as possible. 
The UX and ProjectPage related issues are on going.

**Pull request merged**